[TabrizPlv] iP

TabrizPlv opened the pull request:
# Duke
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Duke is an amazing tool for keeping track of your tasks. It's,
* text-based
* easy to use
* **FAST** to use
All you need to do is,
1. download it from [here](https://github.com/TabrizPlv/ip/releases/tag/A-Jar)
2. double-click it
3. add your tasks
4. let it manage your tasks for you 😃 

And it's ***FREE!***

Features:
- [ ] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

TabrizPlv committed:
Level-1

TabrizPlv committed:
Added ability to store input

TabrizPlv committed:
Added ability to mark and unmark tasks as done

TabrizPlv committed:
Added todos, events and deadlines

TabrizPlv committed:
Text UI Testing

TabrizPlv committed:
Added error handling

TabrizPlv committed:
Added delete function

TabrizPlv committed:
Add save and read from file.
Add date and times

TabrizPlv committed:
Add packages. Apply more OOP

TabrizPlv committed:
Merge remote-tracking branch 'origin/add-gradle-support'

TabrizPlv committed:
Add JUnit tests for Task and ToDo class

TabrizPlv committed:
Add JavaDocs for all classes

TabrizPlv committed:
Adjust to follow Java Coding Standard

TabrizPlv committed:
Add find tasks by keyword

TabrizPlv committed:
Add CheckStyle config

Karen-Lam commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,101 @@
+                throw new DukeExceptions("Description cannot be empty!");
+            }
+        } catch (DukeExceptions e) {
+            System.out.println("Description cannot be empty!");
```
I think using e.getMessage() here to get the Duke exception message written above instead of repeating it twice might be a better option.

Karen-Lam commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,100 @@
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import duke.tasks.*;
```
Imported classes from duke.tasks should be listed explicitly here.

Karen-Lam commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,100 @@
+/** Loads and saves tasks from file*/
+public class Storage {
+    /** User's root directory path*/
+    private static final String userDir = System.getProperty("user.dir");
+    /** Data directory path*/
+    private static final Path dataDirPath = Paths.get(userDir, "data");
+    /** duke.txt file path*/
+    private static final Path dataFilePath = Paths.get(userDir, "data", "duke.Duke.txt");
```
I think that variables declared as final should be in capital letters so that its clearer to read and understand, when referenced to later in the code

Karen-Lam commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,151 @@
+package duke;
+
+import duke.tasks.*;
```
Imported classes from duke.tasks should be listed explicitly here.

Karen-Lam commented on src/main/java/duke/tasks/Event.java:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    /**
+     * Process Event to String to store in duke.txt
```
I think we should use "Processes" here instead of "Process" for the method javadocs

Karen-Lam reviewed (commented):
I think the code looks good! Just some minor violations of the coding standards.

seriouslia0 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,17 @@
+import java.util.Scanner;
+
+public class Duke {
+    public static void main(String[] args) throws DukeExceptions {
```
Amazing use of OOP, very beautiful!

seriouslia0 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,101 @@
+        return new Deadline(desc, date);
+    }
+
+    public static Event parseEvent(String input) {
```
Might want to create subclasses for all the separate parsers

seriouslia0 reviewed (approved):
LGTM

TabrizPlv committed:
Reorganise code to fix all style errors. Add JavaFX

TabrizPlv committed:
Add assert feature in Storage

Used assert in Storage.upload() to make sure data being uploaded is not empty before uploading.

Make GUI changes

Changed the background colours of Button, TextField and ScrollPane. Changed fonts of Button and TextField. Added Prompt message for TextField

TabrizPlv committed:
Improve Code Quality

Reduce the length of Parser methods by abstracting logic into separate methods

TabrizPlv committed:
Merge pull request #1 from TabrizPlv/A-CodeQuality

Improve Code Quality

TabrizPlv committed:
Add User Guide

Added a brief user guide in the ReadMe

Change GUI images

Changed the images for Duke and user in the GUI

TabrizPlv committed:
Edit Duke's responses to commands

TabrizPlv committed:
Minor GUI improvements
Allowed Label to stretch to fit longer lists

TabrizPlv committed:
Update README.md

TabrizPlv committed:
Upload Ui.png

TabrizPlv committed:
Merge branch 'master' of https://github.com/TabrizPlv/ip

TabrizPlv committed:
no message

TabrizPlv committed:
Add ability to detect duplicate Tasks. Extension C-DetectDuplicates.
Tasks that are duplicates of Tasks already existent in the list will not be added.

TabrizPlv committed:
Fix style error

TabrizPlv committed:
bug fix and reformat exceptions

TabrizPlv committed:
bug fix and reformat exceptions
